tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1

```diff
@@ -12,11 +12,89 @@ param(
     [String] $ResourceGroupName,
     [ValidateNotNullOrEmpty()]
     [String] $ClusterName,
+    [String] $CustomLocationOid,
     [Switch] $UseK8s=$false,
     [string] $Tag
 )
 #Requires -RunAsAdministrator
-New-Variable -Name gAksEdgeQuickStartForAioVersion -Value "1.0.240419.1300" -Option Constant -ErrorAction SilentlyContinue
+
+function Verify-ConnectedStatus
+{
+param(
+    [Parameter(Mandatory=$true)]
+    [object] $arcArgs,
+    [Parameter(Mandatory=$true)]
+    [string] $clusterName
+)
+
+    $k8sConnectArgs = @("-g", $arcArgs.ResourceGroupName)
+    $k8sConnectArgs += @("-n", $clusterName)
+    $k8sConnectArgs += @("--subscription", $arcArgs.SubscriptionId)
+
+    $retries = 150
+    for (; $retries -gt 0; $retries--)
+    {
+        $connectedCluster = az connectedk8s show $k8sConnectArgs | ConvertFrom-Json
+        if ($connectedCluster.ConnectivityStatus -eq "Connected")
+        {
+            Write-Host "Cluster reached connected status"
+            break
+        }
+
+        Write-Host "Arc connection status is $($connectedCluster.ConnectivityStatus). Waiting for status to be connected..."
+        Start-Sleep -Seconds 10
+    }
+
+    if ($retries -eq 0)
+    {
+        throw "Arc Connection timed out!"
+    }
+}
+
+function New-ConnectedCluster
+{
+param(
+    [Parameter(Mandatory=$true)]
+    [object] $arcArgs,
+    [Parameter(Mandatory=$true)]
+    [string] $clusterName,
+    [Switch] $useK8s=$false
```

Comment:
is this switch `useK8s` used anywhere?

Reply:
will be used for WorkloadIdentity. passing this in so it is easier to merge later on... WLIF changes are in private branch now...